src/core/main.c

```diff
@@ -1721,17 +1721,58 @@ static void update_numa_policy(bool skip_setup) {
                 log_warning_errno(r, "Failed to set NUMA memory policy: %m");
 }
 
+static void filter_args(const char* dst[], unsigned *pos, char **src, int argc) {
```

Comment:
types are a bit confused here. I mean i understand why argc is an "int", since that is inherited from the main() paameters. But "unsigned *pos" comes from nowhere. it either should also be "int", or it should (preferably I think) be size_t, since this counts emory objects.

Reply:
ah, i see now this is inherited from the stack frame further up, which so far used unsigned.

Reply:
See: #20345